src/commands/project/create-variable-group.ts

```diff
@@ -3,105 +3,112 @@ import commander from "commander";
-    )
-    .option(
-      "--project <project>",
-      "Azure DevOps project name; falls back to azure_devops.project in spk config."
-    )
-    .option(
-      "--personal-access-token <personal-access-token>",
-      "Azure DevOps Personal access token; falls back to azure_devops.access_token in spk config."
-    )
-    .action(async (variableGroupName, opts) => {
-      try {
-        const {
-          registryName,
-          servicePrincipalId,
-          servicePrincipalPassword,
-          tenant
-        } = opts;
-
-        const { azure_devops } = Config();
-
-        const {
-          hldRepoUrl = azure_devops && azure_devops.hld_repository,
-          orgName = azure_devops && azure_devops.org,
-          personalAccessToken = azure_devops && azure_devops.access_token,
-          project = azure_devops && azure_devops.project
-        } = opts;
-
-        const accessOpts: IAzureDevOpsOpts = {
-          orgName,
-          personalAccessToken,
-          project
-        };
-
-        logger.debug(`access options: ${JSON.stringify(accessOpts)}`);
-
-        // required parameters check
-        const errors: string[] = await validateRequiredArguments(
-          variableGroupName,
-          registryName,
-          hldRepoUrl,
-          servicePrincipalId,
-          servicePrincipalPassword,
-          tenant,
-          accessOpts
-        );
-
-        if (errors.length !== 0) {
-          logger.error(
-            `the following arguments are required: ${errors.join("")}`
-          );
-          return errors;
-        }
+export const validateRequiredArguments = (
+  registryName: string | undefined,
+  hldRepoUrl: string | undefined,
+  servicePrincipalId: string | undefined,
+  servicePrincipalPassword: string | undefined,
+  tenant: string | undefined,
+  accessOpts: IAzureDevOpsOpts
+): string[] => {
+  return validateForRequiredValues(decorator, {
+    "hld-repo-url": hldRepoUrl,
+    "org-name": accessOpts.orgName,
+    "personal-access-token": accessOpts.personalAccessToken,
+    project: accessOpts.project,
+    "registry-name": registryName,
+    "service-principal-id": servicePrincipalId,
+    "service-principal-password": servicePrincipalPassword,
+    tenant
+  });
+};
 
+/**
+ * Executes the command.
+ *
+ * @param variableGroupName Variable Group Name
+ * @param opts Option object from command
+ */
+export const execute = async (
+  variableGroupName: string,
+  opts: ICommandOptions,
+  exitFn: (status: number) => void
+) => {
+  if (!hasValue(variableGroupName)) {
+    exitFn(1);
+  } else {
+    try {
+      const { azure_devops } = Config();
+
+      const {
+        registryName,
+        servicePrincipalId,
+        servicePrincipalPassword,
+        tenant,
+        hldRepoUrl = azure_devops && azure_devops.hld_repository,
+        orgName = azure_devops && azure_devops.org,
+        personalAccessToken = azure_devops && azure_devops.access_token,
```

Comment:
Use `?.` operator instead of &&

Reply:
ok. thinking not to change the initial code too much